deploy/0_deploy.ts

```diff
@@ -41,82 +63,186 @@ const ensPublicResolver: PublicResolver = hre.artifacts.require(
+                appRegistryAddress,
+                datasetRegistryAddress,
+                workerpoolRegistryAddress,
+                ethers.constants.AddressZero,
+            )
+            .then((tx) => tx.wait());
+    }
+    // Set categories
+    const catCountBefore = await iexecAccessorsInstance.countCategory();
+    const categories = CONFIG.categories as Category[];
+    for (let i = catCountBefore.toNumber(); i < categories.length; i++) {
+        const category = categories[i];
+        await IexecCategoryManager__factory.connect(erc1538ProxyAddress, owner).createCategory(
+            category.name,
+            JSON.stringify(category.description),
+            category.workClockTimeRef,
+        );
+    }
+    const catCountAfter = await iexecAccessorsInstance.countCategory();
+    console.log(`countCategory is now: ${catCountAfter} (was ${catCountBefore})`);
+    for (let i = 0; i < catCountAfter.toNumber(); i++) {
+        console.log(`Category ${i}: ${await iexecAccessorsInstance.viewCategory(i)}`);
+    }
 };
 
+async function getOrDeployRlc(token: string, owner: SignerWithAddress) {
+    return token // token
+        ? token
+        : await new RLC__factory()
+              .connect(owner)
+              .deploy()
+              .then((contract) => {
+                  contract.deployed();
+                  return contract.address;
+              });
+}
+
+/**
+ * Extract base contract name from contract factory name.
+ * Inputting `MyBoxContract__factory` returns `MyBoxContract`.
+ */
+function getBaseNameFromContractFactory(contractFactory: any) {
+    const name = contractFactory.constructor.name;
+    return name.replace('__factory', '');
+}
+
+/**
+ * Deploy through a GenericFactory a contract [and optionally trigger call]
+ */
+async function deployWithFactory(
+    contractFactory: ContractFactory,
+    constructorArgs?: any[],
+    call?: string,
+) {
+    let bytecode = contractFactory.getDeployTransaction(...(constructorArgs ?? [])).data;
+    if (!bytecode) {
+        throw new Error('Failed to prepare bytecode');
+    }
+    let contractAddress = call
+        ? await genericFactoryInstance.predictAddressWithCall(bytecode, salt, call)
+        : await genericFactoryInstance.predictAddress(bytecode, salt);
+    const previouslyDeployed = (await ethers.provider.getCode(contractAddress)) !== '0x';
+    if (!previouslyDeployed) {
+        call
+            ? await genericFactoryInstance
+                  .createContractAndCall(bytecode, salt, call)
+                  .then((tx) => tx.wait())
+            : await genericFactoryInstance.createContract(bytecode, salt).then((tx) => tx.wait());
+    }
+    const contractName = getBaseNameFromContractFactory(contractFactory);
+    console.log(
+        `${contractName}: ${contractAddress} ${previouslyDeployed ? ' (previously deployed)' : ''}`,
+    );
+    await deployments.save(contractName, {
+        abi: (contractFactory as any).constructor.abi,
```

Comment:
Not very important but if it's not used we can also remove it. It's up to you.

Reply:
Added a comment to be sure: https://github.com/iExecBlockchainComputing/PoCo/pull/58/commits/60b6907887659cbb667a151b3846bf78489d3b70